rclpy/rclpy/node.py

```diff
@@ -648,27 +651,12 @@ def destroy_node(self) -> bool:
         # It will be destroyed with other publishers below.
         self._parameter_event_publisher = None
 
-        with self.handle as capsule:
-
-            while self.publishers:
-                pub = self.publishers.pop()
-                _rclpy.rclpy_destroy_node_entity(pub.publisher_handle, capsule)
-            self.subscriptions = ()
-            while self.clients:
-                cli = self.clients.pop()
-                _rclpy.rclpy_destroy_node_entity(cli.client_handle, capsule)
-            while self.services:
-                srv = self.services.pop()
-                _rclpy.rclpy_destroy_node_entity(srv.service_handle, capsule)
-            while self.timers:
-                tmr = self.timers.pop()
-                _rclpy.rclpy_destroy_entity(tmr.timer_handle)
-                # TODO(sloretz) Store clocks on node and destroy them separately
-                _rclpy.rclpy_destroy_entity(tmr.clock._clock_handle)
-            while self.guards:
-                gc = self.guards.pop()
-                _rclpy.rclpy_destroy_entity(gc.guard_handle)
-
+        self.publishers = ()
```

Comment:
It's a little strange that `self.publishers` (and the rest below) are declared as `List` in the constructor, but are then set to empty tuples here.  I would have guessed mypy would complain about that, but maybe not.  Is this intended, or just an oversight?

Reply:
Calls `.clear()` to remove the elements in the lists instead of setting them to an empty tuple in 4bcfe99